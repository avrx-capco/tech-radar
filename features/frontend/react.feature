Feature: React content page

Background: Navigate to the home page
    Given the user navigates to "http://localhost:3000/"

Scenario: React content appears as expected
    When the user clicks "View Our Tech Radar"
    And the user clicks the hamburger menu
    And the user navigates to "Frontend" via the hamburger menu
    And the user navigates to category "React"
    And the user should see "React is the world’s most popular JavaScript library that uses components to change content of a single-page Mobile and Web application without refreshing."
    And the user should see "Managing states is one of the biggest challenges in large web applications. Using React alone may not be sufficient to handle such complexity. We need a way to share data across components by creating a central store that holds all states of your application."
    And the user should see "The first purpose of testing is to prevent bugs from releasing to production. Testing ensures the features are functioning correctly for complex components and modular applications and is required for the effective performance of a software application or product making it more robust and less prone to error."
    And the user should see "A React, TypeScript web application that provides a more readable format of our Engineering Competency Framework to help explain performance expectations. Unit tests are written in Jest and React Testing Library, E2E tests are written in Cypress and are automated via GitHub Actions."
    And the user clicks accordion "Redux"
    And the user can see accordion "Redux" description "Redux provides a central store for states that each component can access. Main building parts: store, actions, reducers and optional middleware."
    And the user clicks accordion "ContextAPI"
    And the user can see accordion "ContextAPI" description "Context provides a way to pass data through the component tree without having to pass props down manually at every level and helps to avoid the prop-drilling problem. It has 3 main building parts: context, provider and consumer."
    And the user clicks accordion "MaterialUI"
    And the user can see accordion "MaterialUI" description "Google’s Material Design is a popular React UI library with responsive animations and it has been adopted by the wider design community."
    And the user clicks accordion "SemanticUI"
    And the user can see accordion "SemanticUI" description "SemanticUI contains pre-built semantic components that helps create responsive layouts."
    And the user clicks accordion "Ant Design of React"
    And the user can see accordion "Ant Design of React" description "Ant Design of React contains a set of high-quality components and demos for building rich, interactive user interfaces."
    And the user clicks accordion "CSS and SASS"
    And the user can see accordion "CSS and SASS" description "This involves using separate stylesheets like our conventional way of styling our HTML websites either with CSS or a CSS preprocessor called SASS."
    And the user clicks accordion "BEM - Block Element Modifier"
    And the user can see accordion "BEM - Block Element Modifier" description "It's a CSS naming convention for writing cleaner and more readable CSS classes. BEM aims to write independent CSS blocks increasing reusability."
    And the user clicks accordion "CSS Modules"
    And the user can see accordion "CSS Modules" description "A CSS Module is a CSS file in which all class names and animation names are scoped locally by default. The key words here are scoped locally. With this, your CSS class names become similar to local variables in JavaScript. It goes into the compiler, and CSS comes out the other side."
    And the user clicks accordion "Styled Components"
    And the user can see accordion "Styled Components" description "styled-components is a library for React and React Native that allows you to use component-level styles in your application that are written with a mixture of JavaScript and CSS using a technique called CSS-in-JS."
    And the user clicks accordion "Jest"
    And the user can see accordion "Jest" description "Jest is an open-source testing framework. It comes with built-in mocking and assertion libraries, which also runs your tests concurrently in parallel, providing faster test execution."
    And the user clicks accordion "Enzyme"
    And the user can see accordion "Enzyme" description "Enzyme is an open-source JavaScript testing utility by Airbnb and needs a testing library to provide a foundation for the tests it generates. Using Jest and Enzyme together results in cleaner testing code that’s also easier to debug when a test breaks. Jest as a test runner and assertion library, then use Enzyme to build the tests for their UI."
    And the user clicks accordion "React Testing Library"
    And the user can see accordion "React Testing Library" description "The React Testing Library is a very lightweight solution for testing React components. It provides light utility functions on top of react-dom and react-dom/test-utils, in a way that encourages better testing practices."
    And the user clicks accordion "Cypress"
    And the user can see accordion "Cypress" description "Cypress is a JavaScript based end-to-end testing framework that doesn't use Selenium at all. It is built on top of Mocha, which is a feature-rich JavaScript test framework, making asynchronous testing simple. Cypress also uses a BDD/TDD assertion library and a browser that can be paired with any JavaScript testing framework."