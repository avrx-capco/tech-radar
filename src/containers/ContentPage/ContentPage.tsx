import React, { useContext, useEffect, useState } from "react";
import Grid from "@material-ui/core/Grid";
import {
  RadarContextType,
  RadarContext,
} from "../../components/RadarContextProvider/RadarContextProvider";
import BackLink from "../../components/BackLink/BackLink";
import { icons, technologies } from "../../data/data";
import {
  SubContent,
  ExampleContent,
  ReferenceContent,
} from "../../components/Content/Content";
import { TechContentType } from "../../data/content";
import techContent from "../../data/content/index";
import images from "../../images";
import {
  Wrapper,
  ContentWrapper,
  ContentBody,
  Title,
  Divider,
  StyledWrapper,
} from "./styles/";
import ContentNavButton from "../../components/ComponentNavButton/ContentNavButton";
import {
  getNextItem,
  getPrevItem,
  filterClientProjects,
} from "../../helpers/helpers";
import ClientProjectLink from "../../components/ProjectPageLink/ProjectPageLink";
import Axios from "axios";
import ClientProjects from "../../data/projects/ClientProjects";

const CategoryPage = () => {
  const { category, technology, setCategory, setTechnology } =
    useContext<RadarContextType>(RadarContext);
  const [content, setContent] = useState<TechContentType | null>(null);
  const [imageLink, setImageLink] = useState("");
  const [ossProjectCount, setOssProjectCount] = useState<number>(0);
  const [clientProjectCount, setClientProjectCount] = useState<number>(0);

  const handleClick = (categoryName: string, techName: string) => {
    setCategory(categoryName);
    setTechnology(techName);
  };

  const fetchOssProject = async (techName: string) => {
    const response = await Axios.get(
      "https://api.github.com/orgs/capcodigital/repos"
    );
    const data = await response.data;
    const results = data.filter(({ topics }: any) => topics.includes(techName));
    setOssProjectCount(results.length);
  };

  const getClientProjects = (techName: string) => {
    const results = filterClientProjects(ClientProjects, techName);
    setClientProjectCount(results.length);
  };

  useEffect(() => {
    let url = window.location.pathname.split("/");

    // Find category and tech name in technologies data
    let categoriesAndTechnologies =
      technologies.filter(
        ({ categoryName }) =>
          url[2] && categoryName.toLowerCase() === url[2].replace(/-/g, " ")
      )[0] || technologies[0];

    let technologyFromUrl =
      categoriesAndTechnologies.technologies.filter(
        (tech: string) =>
          url[3] && tech.toLowerCase() === url[3].replace(/-/g, " ")
      )[0] || technologies[0].technologies[0];

    let categoryFromUrl = categoriesAndTechnologies.categoryName;

    let icon = icons.filter(
      (el: any) => el.name.toLowerCase() === technologyFromUrl.toLowerCase()
    )[0]!;
    // Filter the content for specific tech
    let content = techContent.filter(
      ({ technology }) => technology === technologyFromUrl
    )[0];

    setContent(content);
    setImageLink(icon.link);
    setTechnology(technologyFromUrl);
    setCategory(categoryFromUrl);
    fetchOssProject(technologyFromUrl.toLowerCase());
    getClientProjects(technologyFromUrl.toLowerCase());

    window.scroll({
      top: 0,
      behavior: "smooth",
    });
  }, [technology, category, setCategory, setTechnology]);

  let nextTechnology = getNextItem(technology);

  let previousTechnology = getPrevItem(technology);

  let filteredNext =
    technology &&
    technologies.filter(({ technologies }) =>
      technologies.includes(nextTechnology)
    )[0];
  let nextCategory = filteredNext ? filteredNext.categoryName : "Devops";

  let filteredPrev =
    technology &&
    technologies.filter(({ technologies }) =>
      technologies.includes(previousTechnology)
    )[0];
  let previousCategory = filteredPrev ? filteredPrev.categoryName : "Mobile";

  return (
    <Wrapper category={category}>
      <img
        className="background"
        src={(images as any)[category]}
        alt={category}
        width={650}
        height={650}
      />
      {content && (
        <ContentWrapper>
          <Grid container justify="space-between">
            <Grid item>
              <BackLink category={category} />
            </Grid>
            <Grid item>
              <StyledWrapper showMobileLink>
                <ClientProjectLink
                  onClick={() => handleClick(category, technology)}
                />
              </StyledWrapper>
            </Grid>
          </Grid>
          <div className="content-head">
            <div>
              <Title className={`title-${technology}`}>{technology}</Title>
              <a href={content.docsLink} target="_blank" rel="noreferrer">
                <svg className="icon" viewBox={"0 0 80 80"}>
                  <circle cx={40} cy={40} r={40} fill={"white"} />
                  <image
                    x={15}
                    y={15}
                    href={imageLink}
                    height={50}
                    width={50}
                  />
                </svg>
              </a>
            </div>
            {(ossProjectCount !== 0 || clientProjectCount !== 0) && (
              <StyledWrapper>
                <ClientProjectLink
                  onClick={() => handleClick(category, technology)}
                />
              </StyledWrapper>
            )}
          </div>
          <ContentBody>
            <div
              className="content-intro"
              dangerouslySetInnerHTML={{ __html: content.intro }}
            />
            <SubContent contentData={content.content} />
            {content.examples.length > 0 && (
              <ExampleContent contentData={content.examples} />
            )}

            {content.reference.length > 0 && (
              <ReferenceContent contentData={content.reference} />
            )}

            <Divider />
            <ContentNavButton
              onClick={() => setTechnology(previousTechnology)}
              previousTechnology={previousTechnology}
              nextTechnology={nextTechnology}
              previousCategory={previousCategory}
              nextCategory={nextCategory}
              next={false}
            />
            <ContentNavButton
              onClick={() => setTechnology(nextTechnology)}
              previousTechnology={previousTechnology}
              nextTechnology={nextTechnology}
              previousCategory={previousCategory}
              nextCategory={nextCategory}
              next={true}
            />
          </ContentBody>
        </ContentWrapper>
      )}
    </Wrapper>
  );
};

export default CategoryPage;
