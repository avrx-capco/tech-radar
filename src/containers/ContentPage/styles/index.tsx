import styled from "styled-components/macro";

type WrapperProps = {
  category: string;
};

export const Wrapper = styled.div<WrapperProps>`
  position: relative;
  width: 100%;
  display: flex;
  padding-left: 90px;
  margin: 50px 0 100px 0;
  .background {
    opacity: 0.12;
    position: absolute;
    top: -80px;
    left: -300px;
  }
  img {
    margin: auto;
  }
  @media screen and (max-width: 1000px) {
    margin-top: 30px;
    padding: 0 25px 0 25px;
    img.background {
      display: none;
    }
  }
`;

export const ContentWrapper = styled.div`
  text-align: left;
  width: 1060px;
  margin: auto;
  position: relative;
  :first-child {
    display: inline-block;
  }
  svg.icon {
    width: 80px;
    height: 80px;
  }
  @media screen and (max-width: 1000px) {
    width: 100%;
    svg.icon {
      width: 50px;
      height: 50px;
    }
  }
  .content-head {
    display: flex;
    justify-content: space-between;
  }
`;

export const ContentBody = styled.div`
  width: 700px;
  margin: 0 auto;
  font-family: poppins, sans-serif;
  .content-intro {
    font-size: 24px;
    margin-bottom: 34px;
    a {
      color: inherit;
      border-bottom: 1px solid ${({ theme }) => theme.pink};
      text-decoration: none;
    }
  }
  @media screen and (max-width: 1000px) {
    width: 100%;
    .content-intro {
      font-size: 18px;
    }
  }
`;

export const Title = styled.div`
  font-size: 120px;
  font-weight: 700;
  text-align: left;
  display: inline-block;
  vertical-align: middle;
  padding-right: 50px;
  @media screen and (max-width: 1000px) {
    font-size: 52px;
    padding-right: 30px;
  }
`;

export const Divider = styled.hr`
  border: 0;
  height: 1px;
  background-image: linear-gradient(
    to right,
    rgba(255, 255, 255, 0),
    rgba(255, 255, 255, 0.75),
    rgba(255, 255, 255, 0)
  );
`;
