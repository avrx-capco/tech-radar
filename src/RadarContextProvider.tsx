import { createContext, useState, FC } from 'react';

export type RadarContextType = {
  category: string;
  technology: string;

  setCategory: (newCategory: string) => void;
  setTechnology: (newComptenecy: string) => void;
};

export const RadarContext = createContext<RadarContextType>({
  category: '',
  technology: '',
  setCategory: (name: string) => {},
  setTechnology: (name: string) => {},
});

export const RadarContextProvider: FC = ({ children }) => {
  const [category, setCategory] = useState('');
  const [technology, setTechnology] = useState('');
  const initialValue = {
    category,
    technology,
    setCategory,
    setTechnology,
  };

  return (
    <RadarContext.Provider value={initialValue}>
      {children}
    </RadarContext.Provider>
  );
};
