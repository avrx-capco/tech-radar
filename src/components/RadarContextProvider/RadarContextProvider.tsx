/* eslint-disable @typescript-eslint/no-empty-function */
import { createContext, useState, FC } from "react";

export type RadarContextType = {
  category: string;
  technology: string;
  setCategory: (newCategory: string) => void;
  setTechnology: (newComptenecy: string) => void;
};

export const RadarContext = createContext<RadarContextType>({
  category: "",
  technology: "",
  setCategory: () => {},
  setTechnology: () => {},
});

export const RadarContextProvider: FC = ({ children }) => {
  const [category, setCategory] = useState("");
  const [technology, setTechnology] = useState("");
  const initialValue = {
    category,
    technology,
    setCategory,
    setTechnology,
  };

  return (
    <RadarContext.Provider value={initialValue}>
      {children}
    </RadarContext.Provider>
  );
};
