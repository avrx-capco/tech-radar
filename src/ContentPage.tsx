import React, { useContext, useEffect, useState } from 'react';
import { RadarContextType, RadarContext } from './RadarContextProvider';
import { Link } from 'react-router-dom';
import styled from 'styled-components/macro';
import { icons, technologies, technologyList } from './data/data';
import images from './images';
import techContent from './data/content';
import { SubContent, ExampleContent } from './Content';
import { TechContentType } from './data/content';
import BackLink from './BackLink';
type WrapperProps = {
  category: string;
};

const Wrapper = styled.div<WrapperProps>`
  position: relative;
  width: 100%;
  display: flex;
  padding-left: 90px;
  margin-top: 50px;
  .background {
    opacity: 0.12;
    position: absolute;
    top: -80px;
    left: -300px;
  }
  img {
    margin: auto;
  }
  hr {
    border: 0;
    height: 1px;
    background-image: linear-gradient(
      to right,
      rgba(255, 255, 255, 0),
      rgba(255, 255, 255, 0.75),
      rgba(255, 255, 255, 0)
    );
  }
  .conent-nav {
  }
`;

const StyledContent = styled.div`
  text-align: left;
  width: 1060px;
  margin: auto;
  position: relative;
  :first-child {
    display: inline-block;
  }
`;

const Title = styled.div`
  font-size: 120px;
  font-weight: 700;
  text-align: left;
  display: inline-block;
  vertical-align: middle;
  padding-right: 50px;
`;

const ContentBody = styled.div`
  width: 700px;
  margin: 0 auto;
  font-family: poppins, sans-serif;
  .content-intro {
    font-size: 24px;
    margin-bottom: 34px;
    a {
      color: inherit;
      border-bottom: 1px solid #e6236d;
    }
  }
`;
type ContentNavButtonProps = {
  align: string;
};
const ContentNavButton = styled.div<ContentNavButtonProps>`
  margin-top: 75px;
  float: ${({ align }) => align};
  div:first-child {
    font-family: poppins, sans-serif;
    font-size: 19px;
    font-weight: 300;
    opacity: 0.5;
  }
  a:nth-child(2) {
    :hover {
      text-decoration: underline;
    }
    font-family: bebas-neue-pro, sans-serif;
    font-size: 59px;
    color: inherit;
  }
`;
const getNextItem = (currentItem: string) => {
  const currentIndex = technologyList.indexOf(currentItem);
  const nextIndex = (currentIndex + 1) % technologyList.length;
  return technologyList[nextIndex];
};
const getPrevItem = (currentItem: string) => {
  const currentIndex = technologyList.indexOf(currentItem);
  const nextIndex = (currentIndex - 1) % technologyList.length;
  return technologyList[nextIndex];
};

const CategoryPage = () => {
  const { category, technology, setCategory, setTechnology } =
    useContext<RadarContextType>(RadarContext);

  const [content, setContent] = useState<TechContentType | null>(null);
  const [imageLink, setImageLink] = useState('');

  useEffect(() => {
    let url = window.location.pathname.split('/');

    //search for the category and tech name in technologies data
    let categoriesAndTechnologies = technologies.filter(
      ({ categoryName }) => categoryName.toLowerCase() === url[2]
    )[0];

    let technologyFromUrl = categoriesAndTechnologies.technologies.filter(
      (tech: string) => tech.toLowerCase() === url[3].replace(/-/g, ' ')
    )[0];

    let categoryFromUrl = categoriesAndTechnologies.categoryName;

    let icon = icons.filter(
      (el: any) => el.name.toLowerCase() === technologyFromUrl.toLowerCase()
    )[0]!;

    // let content = techContent.filter(
    //   ({ technology }) => technology === technologyFromUrl
    // )[0];

    setImageLink(icon.link);
    setTechnology(technologyFromUrl);
    setCategory(categoryFromUrl);
    // setContent(content);
    setContent(techContent[0]);

    window.scroll({
      top: 0,
      behavior: 'smooth',
    });
  }, [technology, category, setCategory, setTechnology]);

  return (
    <>
      <Wrapper category={category}>
        <img
          className='background'
          src={(images as any)[category]}
          alt={category}
          width={650}
          height={650}
        />
        {content && (
          <StyledContent>
            <div>
              <BackLink category={category} />
              <Title>{technology}</Title>
              <a href={content.docsLink} target='_blank'>
                <svg width={80} height={80}>
                  <circle cx={40} cy={40} r={40} fill={'white'} />
                  <image
                    x={15}
                    y={15}
                    href={imageLink}
                    height={50}
                    width={50}
                  />
                </svg>
              </a>
              <ContentBody>
                <div
                  className='content-intro'
                  dangerouslySetInnerHTML={{ __html: content.intro }}
                />
                <SubContent contentData={content.content} />
                <ExampleContent contentData={content.examples} />
                <hr />
                <div className='content-nav'>
                  <ContentNavButton
                    align={'left'}
                    onClick={() => setTechnology(getPrevItem(technology))}
                  >
                    <div>Previous</div>
                    <Link
                      to={`/technology/${category}/${getPrevItem(
                        technology
                      ).replace(/\s/g, '-')}`.toLowerCase()}
                    >
                      {getPrevItem(technology)}
                    </Link>
                  </ContentNavButton>
                  <ContentNavButton
                    align={'right'}
                    onClick={() => setTechnology(getNextItem(technology))}
                  >
                    <div>Next</div>
                    <Link
                      to={`/technology/${category}/${getNextItem(
                        technology
                      ).replace(/\s/g, '-')}`.toLowerCase()}
                    >
                      {getNextItem(technology)}
                    </Link>
                  </ContentNavButton>
                </div>
              </ContentBody>
            </div>
          </StyledContent>
        )}
      </Wrapper>
    </>
  );
};

export default CategoryPage;
