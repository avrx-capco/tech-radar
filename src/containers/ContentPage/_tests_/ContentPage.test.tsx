import { render } from "@testing-library/react";
import { RadarContextProvider } from "components/RadarContextProvider/RadarContextProvider";
import ContentPage from "../ContentPage";
import { MemoryRouter } from "react-router-dom";
import mockAxios from "jest-mock-axios";

window.scroll = jest.fn();

describe("ContentPage", () => {
  beforeEach(() => {
    mockAxios.get.mockResolvedValue({ data: [] });
  });
  afterEach(() => {
    jest.restoreAllMocks();
    mockAxios.reset();
  });

  it("should render ContentPage component", () => {
    const { container } = render(
      <MemoryRouter>
        <RadarContextProvider>
          <ContentPage />
        </RadarContextProvider>
      </MemoryRouter>
    );
    expect(mockAxios.get).toHaveBeenCalledWith(
      "https://api.github.com/orgs/capcodigital/repos"
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
