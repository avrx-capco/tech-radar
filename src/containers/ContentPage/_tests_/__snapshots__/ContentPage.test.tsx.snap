// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentPage should render ContentPage component 1`] = `
.c6 {
  margin-bottom: 40px;
}

.c6 .intro {
  font-size: 18px;
  font-weight: 500;
  padding-bottom: 14px;
}

.c6 .content {
  font-size: 16px;
  font-weight: 300;
}

.c6 .content div:first-child {
  padding: 3px;
  margin: 2px;
  width: 100%;
}

.c6 .content .content-title {
  font-size: 18px;
  font-weight: 700;
}

.c6 a {
  color: inherit;
  border-bottom: 1px solid;
}

.c6 img {
  padding: 20px 0;
}

.c0 {
  position: relative;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding-left: 90px;
  margin: 50px 0 100px 0;
}

.c0 .background {
  opacity: 0.12;
  position: absolute;
  top: -80px;
  left: -300px;
}

.c0 img {
  margin: auto;
}

.c1 {
  text-align: left;
  width: 1060px;
  margin: auto;
  position: relative;
}

.c1:first-child {
  display: inline-block;
}

.c1 svg.icon {
  width: 80px;
  height: 80px;
}

.c1 .content-head {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c5 {
  width: 700px;
  margin: 0 auto;
  font-family: poppins,sans-serif;
}

.c5 .content-intro {
  font-size: 24px;
  margin-bottom: 34px;
}

.c5 .content-intro a {
  color: inherit;
  border-bottom: 1px solid;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c4 {
  font-size: 120px;
  font-weight: 700;
  text-align: left;
  display: inline-block;
  vertical-align: middle;
  padding-right: 50px;
}

.c7 {
  border: 0;
  height: 1px;
  background-image: linear-gradient( to right,rgba(255,255,255,0),rgba(255,255,255,0.75),rgba(255,255,255,0) );
}

.c2 > a {
  font-size: 17px;
}

.c8 {
  margin-top: 75px;
  float: left;
}

.c8 div:first-child {
  font: 300 19px poppins,sans-serif;
  opacity: 0.5;
}

.c8 a:nth-child(2) {
  font: 59px bebas-neue-pro,sans-serif;
  color: inherit;
}

.c8 a:nth-child(2):hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c9 {
  margin-top: 75px;
  float: right;
}

.c9 div:first-child {
  font: 300 19px poppins,sans-serif;
  opacity: 0.5;
}

.c9 a:nth-child(2) {
  font: 59px bebas-neue-pro,sans-serif;
  color: inherit;
}

.c9 a:nth-child(2):hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c3 {
  font-size: 40px;
  font-weight: 700;
  font-family: Bebas-Neue-Pro,sans-serif;
  color: inherit;
  border-bottom: 2px solid;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c3 img {
  margin-left: 10px !important;
}

.c3:hover {
  color: inherit;
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

@media screen and (max-width:1000px) {
  .c6 {
    margin-bottom: 20px;
  }

  .c6 .title {
    font-size: 20px;
  }

  .c6 .intro,
  .c6 .content {
    font-size: 18px;
  }
}

@media screen and (max-width:1000px) {
  .c0 {
    margin-top: 30px;
    padding: 0 25px 0 25px;
  }

  .c0 img.background {
    display: none;
  }
}

@media screen and (max-width:1000px) {
  .c1 {
    width: 100%;
  }

  .c1 svg.icon {
    width: 50px;
    height: 50px;
  }

  .c1 .content-head {
    padding: 30px 0;
  }
}

@media screen and (max-width:1000px) {
  .c5 {
    width: 100%;
  }

  .c5 .content-intro {
    font-size: 18px;
  }
}

@media screen and (max-width:1000px) {
  .c4 {
    font-size: 52px;
    padding-right: 30px;
  }
}

@media screen and (min-width:1000px) {
  .c2 {
    display: none;
  }
}

@media screen and (max-width:1000px) {
  .c8 div:first-child {
    font-size: 19px;
  }

  .c8 a:nth-child(2) {
    font-size: 40px;
  }
}

@media screen and (max-width:1000px) {
  .c9 div:first-child {
    font-size: 19px;
  }

  .c9 a:nth-child(2) {
    font-size: 40px;
  }
}

<div
  class="c0"
>
  <img
    alt="DevOps"
    class="background"
    height="650"
    src="devops.svg"
    width="650"
  />
  <div
    class="c1"
  >
    <div
      class="MuiGrid-root MuiGrid-container css-1xzqqmo-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
      >
        <a
          class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1yxot5x-MuiTypography-root-MuiLink-root"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="NavigateBeforeIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.41 7.41 14 6l-6 6 6 6 1.41-1.41L10.83 12z"
            />
          </svg>
          Back to 
          DevOps
          <img
            alt="DevOps"
            height="24"
            src="devops.svg"
            width="24"
          />
        </a>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
      >
        <div
          class="c2"
        >
          <a
            class="c3"
            href="/projects/devops/github"
          >
            View Projects
          </a>
        </div>
      </div>
    </div>
    <div
      class="content-head"
    >
      <div
        class="MuiBox-root css-70qvj9"
      >
        <div
          class="c4 title-GitHub"
        >
          GitHub
        </div>
        <a
          href="https://docs.github.com/en"
          rel="noreferrer"
          target="_blank"
        >
          <svg
            class="icon"
            viewBox="0 0 80 80"
          >
            <circle
              cx="40"
              cy="40"
              fill="white"
              r="40"
            />
            <image
              height="50"
              href="https://raw.githubusercontent.com/capcodigital/presentation-logos/master/tech-radar/devops/github_square.png"
              width="50"
              x="15"
              y="15"
            />
          </svg>
        </a>
      </div>
    </div>
    <div
      class="c5"
    >
      <div
        class="content-intro"
      >
        GitHub is a cloud-based hosting service that lets you manage Git repositories, Github is ubiquitously used in open source sofrware teams and very prominent within Enterprise, Github provided web based features, team manament and other capbilties on top of Git.
      </div>
      <div
        class="c6"
      />
      <hr
        class="c7"
      />
      <div
        class="c8"
        data-test-id="Previous"
      >
        <div>
          Previous
        </div>
        <a
          href="/technology/data-engineering/liquibase"
        >
          Liquibase
        </a>
      </div>
      <div
        class="c9"
        data-test-id="Next"
      >
        <div>
          Next
        </div>
        <a
          href="/technology/devops/circleci"
        >
          CircleCI
        </a>
      </div>
    </div>
  </div>
</div>
`;
